### PR TITLE
Add previous namespace mismatch warning

### PR DESCRIPTION
**Pull request by Lookyan**
For now we can add at least warning for this issue #3037 
What do you think? May be we can change target namespace if previous release was deleted without purge flag.

### CODE REVIEW

**Comment by thomastaylor312** [comment]
I'll take a look at this in the beginning of next week

**Comment by thomastaylor312** [comment]
@Lookyan So I tried to get this warning to print, but I couldn't get it. `--namespace` is only used if you are in `install`, in which case the release would fail anyway because it already exists. Am I forgetting a use case with this?

**Comment by Lookyan** [comment]
@thomastaylor312 Yes, `--namespace` flag works only with `--install`. And this warning also appears only in case we have `--install` flag and namespaces are not equal. This is the main check for `--namespace` functionality which we have for now.

If we want to have an ability to use this flag without `--install`, algo can be smth like this:
1. Get previous namespace
2. Check whether previous release is in deleted phase or not
3. Compare previous and current namespaces. If they are equal we continue deploying. If they are not equal and previous release is not in deleted phase, we throw warning and exit. And last case: if previous release was deleted, we deploy new release to namespace from `--namespace` flag.

But for now `--namespace` is used only for installation and at least we should have warning from this PR.

So, I can implement this algo if you think it's OK

**Comment by thomastaylor312** [comment]
So I don't think we need to add more to the algorithm, but I was wondering how to trigger this case. I tried doing a `helm upgrade --install --namespace a-different-one ...` and the error didn't print.

**Comment by mcwienczek** [comment]
@thomastaylor312 you need to install, delete the release and then try to install it again using `helm upgrade --install` flag.

**Comment by Lookyan** [comment]
Yes, to get this warning you should do smth like this:
`helm upgrade --install --namespace first myrelease .`
`helm upgrade --install --namespace second myrelease .`

Second command will have this warning because namespace still will be "first" for deployment and "second" won't be applied.

File: cmd/helm/upgrade.go [hunk 1]
```diff
@@ -151,7 +151,15 @@ func (u *upgradeCmd) run() error {
 		// The returned error is a grpc.rpcError that wraps the message from the original error.
 		// So we're stuck doing string matching against the wrapped error, which is nested somewhere
 		// inside of the grpc.rpcError message.
-		_, err := u.client.ReleaseHistory(u.release, helm.WithMaxHistory(1))
+		releaseHistory, err := u.client.ReleaseHistory(u.release, helm.WithMaxHistory(1))
+
+		if err != nil {
+			previousReleaseNamespace := releaseHistory.Releases[0].Namespace
```
Comment (thomastaylor312) [inline cmd/helm/upgrade.go]: This causes a panic if running `helm upgrade --install` because the `releaseHistory` doesn't yet exist

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]

File: cmd/helm/upgrade.go (hunk 1 above)
Comment (Lookyan) [reply cmd/helm/upgrade.go]: Oh, yes, fixed it.

**Review by thomastaylor312 — APPROVED** [review]
Thanks for the fix! This looks good now

**Pull request merged**